infra/modules/providers/azure/app-service/main.tf

```diff
@@ -1,7 +1,10 @@
+data "azurerm_client_config" "current" {}
+
 locals {
   access_restriction_description = "blocking public traffic to app service"
   access_restriction_name        = "vnet_restriction"
   acr_webhook_name               = "cdwebhook"
+  tenant_id                      = var.external_tenant_id == "" ? data.azurerm_client_config.current.tenant_id : var.external_tenant_id
```

Comment:
Recommend that you add locals that looks something like this:
```hcl
app_names  = keys(var.app_service_config)
app_configs = values(var.app_service_config)
```

I think this will improve readability throughout the module. I'll call out areas where it may help in the PR

Reply:
Brilliant Jerry! Brilliant!